### PR TITLE
Chainlink stop-limit orders

### PR DESCRIPTION
**Pull request by mzhu25**
## Description
Implements stop-limit orders at a StaticCall target, using Chainlink as an on-chain price oracle.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/integrations/contracts/src/ChainlinkStopLimit.sol [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        int256 latestPrice = IChainlinkAggregator(oracle).latestAnswer();
+        require(
+            latestPrice >= stopPrice && latestPrice <= limitPrice,
+            "ChainlinkStopLimitOracle/OUT_OF_PRICE_RANGE"
+        );
+
+        uint256 latestTimestamp = IChainlinkAggregator(oracle).latestTimestamp();
+        require(
+            now.safeSub(latestTimestamp) <= priceFreshness, // solhint-disable-line not-rely-on-time
```
[the remark this answers is not in the document]
Comment (abandeali1) [reply contracts/integrations/contracts/src/ChainlinkStopLimit.sol]: This is a pretty interesting parameter. In what scenarios do you see it being most useful?

[the remark this answers is not in the document]
Comment (mzhu25) [reply contracts/integrations/contracts/src/ChainlinkStopLimit.sol]: I can imagine arb bots monitoring the mempool for updates to the aggregator's price, which would serve as an indicator for arb opportunities on these stop-limit orders. The price freshness parameter would mitigate this was my thought

File: contracts/integrations/contracts/src/ChainlinkStopLimit.sol [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "./interfaces/IChainlinkAggregator.sol";
+
+
+contract ChainlinkStopLimit {
+
+    using LibSafeMath for uint256;
```
Comment (abandeali1) [inline contracts/integrations/contracts/src/ChainlinkStopLimit.sol]: Looks like we don't use this anymore.

File: contracts/integrations/src/chainlink_utils.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        { name: 'minPrice', type: 'int256' },
+        { name: 'maxPrice', type: 'int256' },
+    ]);
+    return encoder.encode({ oracle, minPrice, maxPrice });
+}
+/**
+ * Encodes the given stop limit data parameters into StaticCall asset data so that it can be used
+ * in a 0x order.
+ */
+export function encodeStopLimiStaticCallData(
```
Comment (hysz) [inline contracts/integrations/src/chainlink_utils.ts]: `encodeStopLimiStaticCallData` ⟶ `encodeStopLimitStaticCallData` 🇨🇦

**Pull request merged**